### DISCUSSION TITLE
Can't search for topk hits; only getting back data in order it was inserted

### DISCUSSION THREAD

**Discussion by mitchgillin** [milvus-io/milvus#29330] [Q&A and General discussion]
I am using the latest version of Milvus (2.3) and using the NodeSDK. Whenever I run a vector query, I get back the `limit` of my data but there is no sorting by distance or relevance. Eg. If I have `apple`,`banana`, `kiwi` inserted, I will always get back that list in that order. 

 ```
 const query = await client.query({
    collection_name,
    vector: queryVector,
    filter: null,
    limit: 2,
    param: searchParams,
    output_fields: ["text"],
    metric_type: MetricType.L2,
    consistency_level: ConsistencyLevelEnum.Strong,
  });

  await client.releaseCollection({ collection_name });
```

**Comment by xiaofan-luan** [comment]
> I am using the latest version of Milvus (2.3) and using the NodeSDK. Whenever I run a vector query, I get back the `limit` of my data but there is no sorting by distance or relevance. Eg. If I have `apple`,`banana`, `kiwi` inserted, I will always get back that list in that order.
> 
> ```
> const query = await client.query({
>    collection_name,
>    vector: queryVector,
>    filter: null,
>    limit: 2,
>    param: searchParams,
>    output_fields: ["text"],
>    metric_type: MetricType.L2,
>    consistency_level: ConsistencyLevelEnum.Strong,
>  });
> 
>  await client.releaseCollection({ collection_name });
> ```

How did you generate your vector? in your result you should have id and score field and core is similarty between your query and returned corpus
